staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -1670,24 +1676,30 @@ func (az *Cloud) reconcileLoadBalancerRule(
 				},
 			})
 		} else if port.Protocol != v1.ProtocolUDP && port.Protocol != v1.ProtocolSCTP {
-			// we only add the expected probe if we're doing TCP
+			var actualPath string
+			var probePort int32
+
+			probePort = port.NodePort
+
 			if probeProtocol == "" {
-				probeProtocol = string(*probeProto)
+				probeProtocol = string(network.ProbeProtocolHTTP)
 			}
-			var actualPath *string
+
 			if !strings.EqualFold(probeProtocol, string(network.ProbeProtocolTCP)) {
+				probePort = lbNodesHealthCheckPort
 				if requestPath != "" {
-					actualPath = to.StringPtr(requestPath)
+					actualPath = requestPath
 				} else {
-					actualPath = to.StringPtr("/healthz")
+					actualPath = "/healthz"
 				}
 			}
```

Comment:
Just thinking about reversing the logic here, would this produce an identical output?
```
actualPath := requestPath
probePort := lbNodesHealthCheckPort

if probeProtocol == "" {
	// If not protocol is defined, we use HTTP
	probeProtocol = string(network.ProbeProtocolHTTP)
}

if actualPath == "" {
	// If no path is specified, use healthz
	actualPath = "/healthz"
}

if strings.EqualFold(probeProtocol, string(network.ProbeProtocolTCP) {
	// TCP must use the nodeport and not have a path set
	actualPath = ""
	probePort = port.NodePort
}
```